File: TeachingManagementPlatform.Api/Models/CurriculumTemplate.cs
namespace TeachingManagementPlatform.Api.Models;

public class CurriculumTemplate
{
    public int Id { get; set; }
    public string Subject { get; set; } = string.Empty;
    public int Grade { get; set; }
    public string? BookSet { get; set; }
    public int? CreatedBy { get; set; }
    public bool IsPublic { get; set; }
    public string? SourceNote { get; set; }
    public int UsageCount { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    // Navigation properties
    public User? Creator { get; set; }
    public ICollection<CurriculumTemplateLesson> Lessons { get; set; } = new List<CurriculumTemplateLesson>();
}

File: TeachingManagementPlatform.Frontend/src/types/curriculumTemplate.ts
export type TemplateType = 'system' | 'personal' | 'community';

export interface CurriculumTemplate {
  id: number;
  subject: string;
  grade: number;
  bookSet?: string | null;
  createdBy?: number | null;
  isPublic: boolean;
  sourceNote?: string | null;
  usageCount: number;
  lessonCount: number;
  createdAt: string;
}

export interface CurriculumTemplateLesson {
  id: number;
  orderIndex: number;
  chapterName?: string | null;
  lessonName: string;
  suggestedPeriods: number;
}

export interface GeneratedLesson {
  id: number;
  name: string;
  orderIndex: number;
}

export interface GenerateFromTemplateRequest {
  templateId: number;
}

export interface GenerateFromTemplateResponse {
  lessons: GeneratedLesson[];
}

export interface SaveAsTemplateRequest {
  isPublic: boolean;
  bookSet?: string;
  sourceNote?: string;
}

export interface LessonItemDto {
  orderIndex: number;
  chapterName?: string;
  lessonName: string;
  suggestedPeriods: number;
}

export interface CreateCurriculumTemplateRequest {
  subject: string;
  grade: number;
  bookSet?: string;
  sourceNote?: string;
  lessons?: LessonItemDto[];
}

export interface UpdateCurriculumTemplateRequest {
  subject?: string;
  grade?: number;
  bookSet?: string;
  sourceNote?: string;
  isPublic?: boolean;
}

export interface BulkUpdateLessonsRequest {
  lessons: LessonItemDto[];
}
